types/index.d.ts

```diff
@@ -135,8 +135,12 @@ declare module 'react-sweet-state' {
     } & TProps
   >;
 
-  type HookFunction<TState, TActions, TArg = undefined> = (
-    ...args: TArg extends undefined ? [] : [TArg]
+  type HookFunction<TState, TActions, TArg = void> = (
+    ...args: TArg extends unknown
+      ? TArg extends undefined
```

Comment:
If it extends to the `unknown` which can be anything, if it extends to undefined then allow `[]` so if nothing gets passed in the caller function, `TS` will not cause a problem. As in JS no param automatically becomes `undefined`.